src/components/ConfigureQuery/ConfigureQuery.tsx

```diff
@@ -48,80 +54,111 @@ export const ConfigureQuery:FC<ConfigureQueryProps> = (
-    } }).json();
+  const testQueryDataSource = async ({ fqlQuery }: { fqlQuery: FqlQuery }) => {
+    return ky
+      .post('query', {
+        json: {
+          entityTypeId: selectedType,
+          fqlQuery: JSON.stringify(fqlQuery),
+        },
+      })
+      .json();
   };
 
-  const runQueryDataSource = ({ fqlQuery, queryId } : { fqlQuery: any, queryId: string }) => {
-    const data = {
-      name: listName,
-      description,
-      fields: columns,
-      isActive: status === STATUS_VALUES.ACTIVE,
-      isPrivate: visibility === VISIBILITY_VALUES.PRIVATE,
-      queryId,
-      version,
-      fqlQuery: JSON.stringify(fqlQuery)
-    };
-
-    if (!isEditQuery) {
-      // @ts-ignore:next-line
-      data.entityTypeId = selectedType;
+  const runQueryDataSource = ({ fqlQuery, queryId }: { fqlQuery: FqlQuery; queryId: string }) => {
+    if (isEditQuery) {
+      const data: ListForUpdate = {
+        name: listName,
+        description,
+        fields: columns,
+        isActive: status === STATUS_VALUES.ACTIVE,
+        isPrivate: visibility === VISIBILITY_VALUES.PRIVATE,
+        queryId,
+        version: version as number,
+        fqlQuery: JSON.stringify(fqlQuery),
+      };
+
+      return ky.put(`lists/${listId}`, { json: data }).json();
+    } else {
+      const data: ListForCreation = {
+        name: listName,
+        description,
+        fields: columns,
+        isActive: status === STATUS_VALUES.ACTIVE,
+        isPrivate: visibility === VISIBILITY_VALUES.PRIVATE,
+        queryId,
+        fqlQuery: JSON.stringify(fqlQuery),
+        entityTypeId: selectedType,
+      };
+
+      return ky.post('lists', { json: data }).json();
     }
-
-    return isEditQuery ? ky.put(`lists/${listId}`, { json: data }).json() : ky.post('lists', { json: data }).json();
   };
 
-  const onQueryRunSuccess = ({ id } : { id: string}) => {
-    showSuccessMessage({ message: t('callout.list.save.success', {
-      listName
-    }) });
+  const onQueryRunSuccess = ({ id }: { id: string }) => {
+    showSuccessMessage({
+      message: t('callout.list.save.success', {
+        listName,
+      }),
+    });
 
     history.push(`${HOME_PAGE_URL}/list/${id}`);
   };
 
   const onQueryRunFail = (error: HTTPError) => {
     (async () => {
       const errorMessage = await computeErrorMessage(error, 'update-optimistic.lock.exception', {
-        listName
+        listName,
       });
 
       showErrorMessage({ message: errorMessage });
     })();
   };
 
-  const getParamsSource = async ({ entityTypeId, columnName, searchValue }: any) => {
-    return ky.get(`entity-types/${entityTypeId}/columns/${columnName}/values?search=${searchValue}`).json();
+  const getParamsSource = async ({
+    entityTypeId,
+    columnName,
+    searchValue,
+  }: {
+    entityTypeId: string;
+    columnName: string;
+    searchValue: string;
+  }) => {
+    return ky
+      .get(`entity-types/${entityTypeId}/columns/${columnName}/values?search=${searchValue}`)
```

Comment:
this was another bug i saw here - if `searchValue` contains something like `foo?bar`, it'll only actually get `foo`. i'm going to spin this off into a separate ticket though

Reply:
spun off to https://folio-org.atlassian.net/browse/UILISTS-124